lib/appsignal/config.rb

```diff
@@ -410,6 +417,33 @@ def maintain_backwards_compatibility(configuration)
       end
     end
 
+    # Maintain backwards compatibility with deprecated config options.
+    #
+    # Add deprecated config options here with the behavior of setting its
+    # replacement, if the replacement option is not configured by the user.
+    #
+    # Make sure to remove the contents of this method in the next major
+    # version, but the method itself with an empty body can stick around as a
+    # structure for future deprecations.
+    def maintain_backwards_compatibility
+      skip_session_data = config_hash[:skip_session_data]
+      send_session_data = config_hash[:send_session_data]
+      if skip_session_data.nil? # Deprecated option is not set
+        if send_session_data.nil? # Not configured by user
+          @system_config[:send_session_data] = true
```

Comment:
I now realise I should have left [the comment on the diagnose tools PR](https://github.com/appsignal/diagnose_tests/pull/59#discussion_r783846011) here instead.

When setting it to the default value, shouldn't it be added to `@initial_config`, instead of `@system_config`? That way it will show up as `initial` in the diagnose tool.

Reply:
The `initial_config` is what the user gives to the initializer of the config class. If I add it to that the initial source it implies that the user set it like this: `Appsignal::Config.new(path, env, { :send_session_data => true })

I've set it to the system because the system (the gem) decides based on several factors (if the deprecated option is set or not) what the value becomes.

Reply:
> If I add it to that the initial source it implies that the user set it like this: `Appsignal::Config.new(path, env, { :send_session_data => true })`

Right, I see -- I got the terms mixed then. I meant to say it should show up as `default`: when it is set to `true` because neither `skip_` or `send_` were set by the user, it is acting as a default value.

I do agree with it being set to system when it is derived from the value of the deprecated option, and I see that there's no `@default_config` instance variable, so that might complicate the implementation of this somewhat 😕. I'd be fine with it either way, I think.

Reply:
> Otherwise it implies that the default value of send_session_data can be true or false, which doesn't make a default value. A config option can only have one default value.

I agree that it should still be coming from the system source when it's determined by the value of `skip_session_data` -- but I was talking about the case in the bit of code highlighted, where both `skip_` and `send_` are unset by the user, and the value of the configuration value is then set to `true` (its default value).